### PR TITLE
Fix instructions for 'python setup.py install' to be 'pip install .' …

### PR DESCRIPTION
**Pull request by brittonsmith**

### CODE REVIEW

File: doc/source/installing.rst [hunk 1]
```diff
@@ -407,7 +407,7 @@ development version of yt instead of the latest stable release, you will need
   $ git clone https://github.com/yt-project/yt
   $ cd yt
   $ git checkout master
-  $ pip install -e . --user --prefix=
+  $ pip install . --user --prefix=
```
Comment (ngoldbaum) [inline doc/source/installing.rst]: I'm not sure offhand whether `--prefix=` can be passed to pip at all. I think you need to do say `--install-option="--prefix="`

I'm also not sure offhand whether `--prefix=` is even necessary anymore, I think it was a workaround for an issue on older fedora installs?

Comment (brittonsmith) [inline doc/source/installing.rst]: Ok, I can change that.

Comment (brittonsmith) [inline doc/source/installing.rst]: Just to be clear, this should read:
```
pip install . --install-option="--user" --install-option="--prefix="
```
Is that right?

Comment (ngoldbaum) [inline doc/source/installing.rst]: ``--user`` doesn't need `--install-option`, it's supported by pip: http://pip.readthedocs.io/en/stable/reference/pip_install/#cmdoption-user

Comment (ngoldbaum) [inline doc/source/installing.rst]: And please also update the notes about why we're using `--user` and `--prefix` in this document.

**Pull request merged**